src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,140 @@
+
+If the user follows the ‘Not received a text message?’ link, allow them to check which mobile number they entered, and to change it if necessary. This prevents the user becoming stuck if they entered a mobile number incorrectly. Do not allow users to change the number when signing in.
+
+{{ example({group: "patterns", item: "security-code", example: "resend-first-time", html: true, nunjucks: true, open: false}) }}
+
+This should send a new code, with its own 15 minute time limit. The previous code should remain valid within its original time limit.
+
+### When signing in
+
+When a user returns to your service, verify their password first. Once they have entered this correctly, send a text message to their mobile phone with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
+}) }}
+
+Ask the user to enter this code. Use the same pattern and time limit as when creating an account.
+
+If they follow the ‘Not received a text message?’ link, show them a page allowing them to request a new code. Do not reveal the mobile number you sent it to.
+
+{{ example({group: "patterns", item: "security-code", example: "resend", html: true, nunjucks: true, open: false}) }}
+
+You should tell users what to do if they no longer have access to the phone used to sign up.
+
+For example, you might tell them to contact a support address.
+
+### Domain-bound codes
+
+You can also follow the [domain-bound codes](https://developer.apple.com/news/?id=z0i801mg) standard. This improves security by restricting the code to a specific domain.
+
+Include the domain of the service on a new line, prefixed with an `@`, followed by a `#` symbol and the code, like this:
+
+{{ govukInsetText({
+html: "12345 is your [service name] security code<br><br>@www.example.service.gov.uk #12345"
+}) }}
+
+### Error messages
+
+Style error messages like this:
+
+{{ example({group: "patterns", item: "security-code", example: "error-incorrect", html: true, nunjucks: true, open: false}) }}
+
+If the user doesn’t enter enough digits:
+<br>Say ‘You've not entered enough numbers, the code must be 5 numbers’.
+
+If the user enters too many digits:
+<br>Say ‘You’ve entered too many numbers, the code must be 5 numbers’.
+
+If the user enters non-numeric characters, other than spaces:
+<br>Say ‘The code must be 5 numbers’.
+
+If the code has expired, show this message:
+
+{{ example({group: "patterns", item: "security-code", example: "error-expired", html: true, nunjucks: true, open: false}) }}
+
+If the code was sent more than 2 hours ago, show the 'Incorrect security code' message.
+### Technical security measures
+
+Use [technical measures to prevent automated attacks](https://www.ncsc.gov.uk/collection/passwords/updating-your-approach#tip2-password-collection), such as adding a time delay after 10 incorrect number entries.
```

Comment:
```suggestion
Use [technical measures to prevent automated attacks](https://www.ncsc.gov.uk/collection/passwords/updating-your-approach#tip2-password-collection). For example, add a time delay after 10 incorrect number entries.
```